test/test_client.py

```diff
@@ -361,6 +363,21 @@ def test_metadata(self):
         )
         options = client.options
         self.assertEqual(options.pool_options.metadata, metadata)
+        # Test truncating driver info metadata.
+        client = MongoClient(driver=DriverInfo(name="s" * _MAX_METADATA_SIZE))
```

Comment:
We need to pass connect=False to MongoClient here and below since this is part of ClientUnitTest: 
```
class ClientUnitTest(unittest.TestCase):
    """MongoClient tests that don't require a server."""
```